openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -31,6 +38,116 @@ def get_identity(self, data):
         except AttributeError:
             return None
 
+class AccountFullUserProfileReadOnlySerializer(serializers.Serializer):
+    """
+    Class that serializes the portion of User model needed for account information.
+    """
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'configuration' arg up to the superclass
+        self.configuration = kwargs.pop('configuration', None)
+        if not self.configuration:
+            self.configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
+
+        # Don't pass the 'admin_fields' arg up to the superclass
+        self.admin_fields = kwargs.pop('admin_fields', False)
+
+        # Instantiate the superclass normally
+        super(AccountFullUserProfileReadOnlySerializer, self).__init__(*args, **kwargs)
+
+    def to_native(self, user):
+        """
+        Overwrite to_native to handle custom logic since we are serializing two models as one here
+        :param user: User object
+        :return: Dict serialized account
+        """
+        profile = user.profile
+
+        data = {
+            "username": user.username,
+            "url": ''.join([
+                'http://',
+                get_current_site(None).domain,
+                reverse('accounts_api', kwargs={'username': user.username})
+            ]),
+            "email": user.email,
+            "date_joined": user.date_joined,
+            "is_active": user.is_active,
+            "bio": profile.bio,
+            "country": profile.country.code,
+            "profile_image": self._get_profile_image(profile, user),
+            "time_zone": None,
+            "language_proficiencies": LanguageProficiencySerializer(
+                profile.language_proficiencies.all(),
+                many=True
+            ).data,
+            "name": profile.name,
+            "gender": profile.gender,
+            "goals": profile.goals,
+            "year_of_birth": profile.year_of_birth,
+            "level_of_education": profile.level_of_education,
+            "mailing_address": profile.mailing_address,
+            "requires_parental_consent": profile.requires_parental_consent(),
+        }
+
+        return self._filter_fields(
+            self._visible_fields(profile, user),
+            data
+        )
+    def _get_profile_image(self, user_profile, user):
+        """ Returns metadata about a user's profile image. """
+        data = {'has_image': user_profile.has_profile_image}
+        urls = get_profile_image_urls_for_user(user)
```

Comment:
I think it would be clearer if ``get_profile_image_urls_for_user`` returned absolute URLs. If there are use cases where this isn't necessary then I'd control it with an optional boolean parameter.